soc/nxp/mcx/Kconfig.defconfig

```diff
@@ -5,4 +5,32 @@ if SOC_FAMILY_NXP_MCX
 
 rsource "*/Kconfig.defconfig"
 
+if FLASH_MCUX_FLEXSPI_XIP
+
+# Avoid RWW hazards by defaulting logging to disabled
+choice FLASH_LOG_LEVEL_CHOICE
+	default FLASH_LOG_LEVEL_OFF
+endchoice
+
+choice MEMC_LOG_LEVEL_CHOICE
+	default MEMC_LOG_LEVEL_OFF
+endchoice
+
+# Code relocation is needed when FLASH_MCUX_FLEXSPI_XIP is enabled
+config CODE_DATA_RELOCATION_SRAM
```

Comment:
This seems like it will turn on code relocation whenever `FLASH_MCUX_FLEXSPI_XIP` is enabled, but it looks like the MCX only actually uses code relocation when the FLEXSPI flash driver is enabled: https://github.com/zephyrproject-rtos/zephyr/blob/main/soc/nxp/mcx/mcxn/CMakeLists.txt#L16, so should this setting depend on `FLASH_MCUX_FLEXSPI_NOR`?

Reply:
no, it's common to all the SOC due to the drivers being relocated https://github.com/zephyrproject-rtos/zephyr/blob/0a7908588c755d3b8dfa3cc5c967d8a085762500/drivers/flash/CMakeLists.txt#L66

Reply:
Ok- it looks like you dropped this definition and moved to the common one, which does have the right dependencies (as far as I can tell)